scraper/ordoclic.py

```diff
@@ -0,0 +1,111 @@
+    r = session.post(base_url, data=json.dumps(payload), headers=headers)
+    r.raise_for_status()
+    return(r.json())
+
+def getProfile(rdv_site_web):
+    profile_name = rdv_site_web.rsplit('/', 1)[-1]
+    base_url = f'https://api.ordoclic.fr/v1/public/entities/profile/{profile_name}'
+    r = session.get(base_url)
+    r.raise_for_status()
+    return(r.json())
+
+def parse_ordoclic_slots(availability_data):
+    first_availability = None
+    if not availability_data:
+        return None
+    if 'nextAvailableSlotDate' in availability_data:
+        nextAvailableSlotDate = availability_data.get('nextAvailableSlotDate', None)
+        if nextAvailableSlotDate != None:
+            first_availability = datetime.strptime(nextAvailableSlotDate, '%Y-%m-%dT%H:%M:%S%z') + timedelta(minutes=120)
+            return first_availability
+
+    availabilities = availability_data.get('slots', None)
+    if availabilities is None:
+        return None
+    for slot in availabilities:
+        timeStart = slot.get('timeStart', None)
+        if not timeStart:
+            continue
+        date = datetime.strptime(timeStart, '%Y-%m-%dT%H:%M:%S%z')
+        if 'timeStartUtcOffset' in slot:
+            timeStartUtcOffset = slot["timeStartUtcOffset"]
+            date += timedelta(minutes=timeStartUtcOffset)
+        if date is None:
+            continue
+        if first_availability is None or date < first_availability:
+                    first_availability = date        
+    return first_availability
+    
+def fetch_centre_slots(rdv_site_web, start_date):
+    first_availability = None    
+    profile = getProfile(rdv_site_web)
+    slug = profile["profileSlug"]
+    entityId = profile["entityId"]
+    for professional in profile["publicProfessionals"]:
+        medicalStaffId = professional["id"]
+        name = professional["fullName"]
+        zip = professional["zip"]
+        reasons = getReasons(entityId)
+        #reasonTypeId = 4 -> 1er Vaccin
+        for reason in reasons["reasons"]:
+            if reason["reasonTypeId"] == 4 and reason["canBookOnline"] == True: 
+                reasonId = reason["id"]
+                date_obj = datetime.strptime(start_date, '%Y-%m-%d')
+                end_date = (date_obj + timedelta(days=6)).strftime('%Y-%m-%d')
+                slots = getSlots(entityId, medicalStaffId, reasonId, start_date, end_date)
+                date = parse_ordoclic_slots(slots)
+                if date is None:
+                    continue
+                if first_availability is None or date < first_availability:
+                    first_availability = date
+    return first_availability
+
+def centre_iterator():   
+    items = search()
+    for item in items["items"]:
+        payload = {}
+        slug = item["publicProfile"]["slug"]
+        rdv_site_web = f"https://app.ordoclic.fr/app/pharmacie/{slug}"
+        start_date = datetime.now().isoformat()[:10]
```

Comment:
Ca fonctionne mais utiliser `strftime('%Y-%m-%d')` ce serait quand même plus propre.
```suggestion
        start_date = datetime.now().strftime("%Y-%m-%d")
```